src/Intl/FranceCitiesBundle.php

```diff
@@ -144,6 +144,22 @@ public static function getCityDataFromInseeCode(string $inseeCode): ?array
         return null;
     }
 
+    /**
+     * Returns the city name for the given INSEE code or null if the city was not found.
+     */
+    public static function getCityNameFromInseeCode(string $inseeCode): ?string
+    {
+        foreach (self::$cities as $postalCode => $cities) {
+            $trimmedInseeCode = ltrim($inseeCode, '0');
+
+            if (array_key_exists($inseeCode, $cities) || array_key_exists($trimmedInseeCode, $cities)) {
```

Comment:
we should try the $trimmedInseeCode in first place, since in 99% of cases it will be this one matching.

Reply:
fixed 👍